components/frontend/src/utils.test.js

```diff
@@ -509,3 +511,18 @@ it("returns whether a measurement is requested for the metric", () => {
     expect(isMeasurementRequested({ measurement_requested: "2023-01-01", latest_measurement: latest })).toBe(false)
     expect(isMeasurementRequested({ measurement_requested: "2025-01-01", latest_measurement: latest })).toBe(true)
 })
+
+it("returns whether a metric's measurement is stale", () => {
+    expect(isMeasurementStale({})).toBe(false)
+    expect(isMeasurementStale({ latest_measurement: { end: "2024-09-23" } })).toBe(true)
+    expect(isMeasurementStale({ latest_measurement: { end: "2024-09-23" } }, new Date("2024-09-24"))).toBe(true)
+    expect(isMeasurementStale({ latest_measurement: { end: "2024-09-24" } }, new Date("2024-09-23"))).toBe(false)
+})
+
+it("returns whether a metric's measurement is outdated", () => {
+    expect(isMeasurementOutdated({})).toBe(false)
+    expect(isMeasurementOutdated({ sources: {} })).toBe(false)
+    expect(isMeasurementOutdated({ sources: { source_uuid: {} } })).toBe(true)
```

Comment:
This test case is a bit magical without explanatory comments

Reply:
I'll add some comments!